Testscripts/Linux/verify_uio_driver.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+########################################################################
+#
+# Description:
+#       This script was created to verify the uio_hv_generic module.
+#
+#       The test performs the following steps:
+#         1. Assign a NIC to uio_hv_generic driver and check the uio device
+#         2. Restore to hv_netvsc and check the eth1
+#
+########################################################################
+
+# Source utils.sh
+. utils.sh || {
+    echo "ERROR: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+echo "device_id=$(cat /sys/class/net/eth1/device/device_id|tr -d '{}')" >> constants.sh
+echo "class_id=$(cat /sys/class/net/eth1/device/class_id|tr -d '{}')" >> constants.sh
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+function checkCmdExitStatus ()
+{
+	exit_status=$?
+	cmd=$1
+	if [ $exit_status -ne 0 ]; then
+		LogErr "$cmd: FAILED (exit code: $exit_status)"
+        [ -n "$2" ]&&LogErr "$2"
+		SetTestStateFailed
+		exit $exit_status
+	else
+		LogMsg "$cmd: SUCCESS" 
+	fi
+}
+
+function register_uio() {
+    LogMsg "Register eth1 to uio"
+    output=$(modprobe uio_hv_generic 2>&1)
+    lsmod|grep uio_hv_generic
+    checkCmdExitStatus "Verify uio_hv_generic is loaded"
+
+    [ -z "$output" ]
+    checkCmdExitStatus "Verify no output while modprobe uio_hv_generic" "Output: $output"
+
+    echo ${class_id} > /sys/bus/vmbus/drivers/uio_hv_generic/new_id
+    echo ${device_id} > /sys/bus/vmbus/drivers/hv_netvsc/unbind
+    echo ${device_id} > /sys/bus/vmbus/drivers/uio_hv_generic/bind
+    # Verify /dev/uio0 exists
+    [ -e /dev/uio0 ]
+    checkCmdExitStatus "Verify /dev/uio0 exists"
```

Comment:
is that any real scenario that we can use uio0 for further scenario validation?

Reply:
Hi @LiliDeng ,
I don't know much but only know the netvsc PMD in DPDK.